src/TraceEvent/Parsers/GCDynamicTraceEventParser.cs

```diff
@@ -64,18 +66,16 @@ protected internal override void EnumerateTemplates(Func<string, string, EventFi
                     callback(template);
         }
 
-        /// <summary>
-        /// Do not use.  This is here to avoid asserts that detect undeclared event templates.
-        /// </summary>
-        public event Action<GCDynamicTraceEvent> GCDynamicTraceEvent
+        private event Action<RawDynamicTraceEvent> _gcDynamicTraceEvent;
+        public event Action<RawDynamicTraceEvent> GCDynamicTraceEvent
```

Comment:
this is a public class, is it exposed to users of TraceEvent at all?

Reply:
Is there a particular concern here about the exposure? Are you worried about the signature change breaking existing apps, or something else?

Reply:
yes, I'm asking if changing a public class would break any existing usage.

Reply:
Earlier I missed the fact that I was changing the generic type argument.

I don't think any "useful" usage could be broken with this change. We can certainly write a contrived scenario such that this change broke something.